File: solutions/30-12943.rb
def solution(num)
  num == 1 ? 0 : (1..500).detect{|i| (num = num.even? ? num/2 : (num*3)+1) && num == 1 } || -1
end

p solution(6)
p solution(16)
p solution(626331)
p solution(1)

File: solutions/py/30-12943.py
def solution(num):
    for n in range(500):
        if num == 1:
            return n
        num = num*3+1 if num % 2 else num / 2
    return -1


print(solution(1)) #=> 0
print(solution(6)) #=> 8
print(solution(16)) #=> 4
print(solution(626331)) #=> -1
